src/api/server.tsx

```diff
@@ -110,6 +110,32 @@ export const fetchResources = async (
     });
 };
 
+export const fetchClusterMembersResources = async (
+  clusterMembers: LxdClusterMember[],
+): Promise<LxdResources[]> => {
+  return new Promise((resolve, reject) => {
+    Promise.allSettled(
+      clusterMembers.map(async (member) => {
+        return fetchResources(member.server_name);
+      }),
+    )
+      .then((results) => {
+        const data: LxdResources[] = [];
+        for (let i = 0; i < clusterMembers.length; i++) {
+          const member = clusterMembers[i];
+          const result = results[i];
+          if (result.status === "fulfilled") {
+            data.push(result.value);
+          } else if (result.status === "rejected") {
+            reject(constructMemberError(result, member.server_name));
+          }
+        }
+        resolve(data);
+      })
+      .catch(reject);
+  });
+};
+
```

Comment:
```suggestion
export const fetchClusterMembersResources = async (
  clusterMembers: LxdClusterMember[],
): Promise<LxdResources[]> => {
  const results = await Promise.allSettled(
    clusterMembers.map(async (member) => fetchResources(member.server_name)),
  );

  const data: LxdResources[] = [];

  for (let i = 0; i < clusterMembers.length; i++) {
    const member = clusterMembers[i];
    const result = results[i];

    if (result.status === "rejected") {
      throw constructMemberError(result, member.server_name);
    }

    data.push(result.value);
  }

  return data;
};
```

I know that the other functions are written like this, but should we use `async`/`await` ?